modules/virt-building-real-time-container-disk-image.adoc

```diff
@@ -3,163 +3,98 @@
-+
-[source,terminal]
-----
-$ cat << EOF > real-time-vm.toml
-name = "realtime_image"
-description = "Image to use with the real-time checkup"
-version = "0.0.1"
-distro = "rhel-87"
-
-[[customizations.user]]
+. Run a {op-system-base-full} 8 Operating system image create a new root user with the following user and password:
 name = "root"
 password = "redhat"
 
-[[packages]]
-name = "real-time"
-
-[[packages]]
-name = "real-time-tools"
-
-[[packages]]
-name = "driverctl"
-
-[[packages]]
-name = "tuned-profiles-cpu-partitioning"
-
-[customizations.kernel]
-append = "default_hugepagesz=1GB hugepagesz=1G hugepages=1"
-
-[customizations.services]
-disabled = ["NetworkManager-wait-online", "sshd"]
-EOF
-----
-
-. Push the blueprint file to the image builder tool by running the following command:
-+
-[source,terminal]
-----
-# composer-cli blueprints push realtime-vm.toml
-----
-
-. Generate the system image by specifying the blueprint name and output file format. The Universally Unique Identifier (UUID) of the image is displayed when you start the compose process.
+. login to this root user and install the following packages:
 +
 [source,terminal]
 ----
-# composer-cli compose start realtime_image qcow2
+# dnf install kernel-rt tuned tuned-profiles-realtime tuned-profiles-nfv-guest rt-tests cloud-init
 ----
-
-. Wait for the compose process to complete. The compose status must show `FINISHED` before you can continue to the next step.
 +
+[NOTE]
+====
+You may need to enable to following repositories:
 [source,terminal]
 ----
-# composer-cli compose status
+# subscription-manager repos --enable rhel-8-for-x86_64-rt-rpms rhel-8-for-x86_64-nfv-rpms
 ----
+====
 
-. Enter the following command to download the `qcow2` image file by specifying its UUID:
-+
+. Disable the following services:
 [source,terminal]
 ----
-# composer-cli compose image <UUID>
+# systemctl disable NetworkManager-wait-online sshd irqbalance
 ----
 
-. Create the customization scripts by running the following commands:
+. Copy this script on the image:
 +
 [source,terminal]
 ----
-$ cat <<EOF >customize-vm
+# cat <<EOF >customize-vm.sh
 #!/bin/bash
 
-# Setup hugepages mount
-mkdir -p /mnt/huge
-echo "hugetlbfs /mnt/huge hugetlbfs defaults,pagesize=1GB 0 0" >> /etc/fstab
+set -xe
 
-# Create vfio-noiommu.conf
-echo "options vfio enable_unsafe_noiommu_mode=1" > /etc/modprobe.d/vfio-noiommu.conf
+# Disable swap
+swapoff -a
+sed -i '/swap/s/^/#/' /etc/fstab
 
 # Enable guest-exec,guest-exec-status on the qemu-guest-agent configuration
 sed -i '/^BLACKLIST_RPC=/ { s/guest-exec-status//; s/guest-exec//g }' /etc/sysconfig/qemu-ga
 sed -i '/^BLACKLIST_RPC=/ { s/,\+/,/g; s/^,\|,$//g }' /etc/sysconfig/qemu-ga
 EOF
 ----
 
-. Use the `virt-customize` tool to customize the image generated by the image builder tool:
+. Run the script on the image:
```

Comment:
I can image that there should be a step to export the VM to a qcow2 image file.
In that case, the steps performed by the script, could be performed inside the guest prior to exporting it, thus making the need to use virt-customize redundant.

Reply:
We're not running virt-customize anymore - I just kept the scipt name `customize-vm.sh`

Reply:
I see.
Since this script requires `root` privileges please explicitly state it.

Reply:
Since the manual is all in root user (the one we created and logged in with on earlier steps) - I'm not sure it's needed. WDYT?